### ISSUE TITLE
Ban mitigations

### ISSUE THREAD

**Issue by cedws** [cedws/discord-delete#33]
Following on from [discussions](https://github.com/victornpb/deleteDiscordMessages/issues/193) in a related project, I am investigating how bans for self-botting could be mitigated. For now, I have implemented UA and super properties spoofing to make the program look more like the official client - I'm waiting to see if this will work.

I suspect that the way Discord's heuristics works is that users with a high number of transactions are subject to special logic which scrutinises requests and checks whether they look like they are coming from the official client.

**Comment by cedws** [comment]
I am thinking of running some A/B tests with a large number of testing accounts to try and find the ban criteria. Can probably be done on AWS Lambda.

**Comment by git-bruh** [comment]
FWIW My account didn't get banned, but restricted behind the phone verification crap recently, I just ran a full deletion and immediately left a guild after that. Can't seem to use this method to get restricted reliably though, maybe it was just a coincidence :/

**Comment by SpheeresX** [comment]
> FWIW My account didn't get banned, but restricted behind the phone verification crap recently, I just ran a full deletion and immediately left a guild after that. Can't seem to use this method to get restricted reliably though, maybe it was just a coincidence :/

Same here. It wasn't a coincidence.

**Comment by SwampyApple** [comment]
Any new insights on this issue?

**Comment by cedws** [comment]
Nope, but I haven't seen any more bans since introduction of some spoofing.

**Comment by lillelilje** [comment]
I'll say that I've been running this script pretty hardcore recently, while also using the desktop client and phone client and I haven't faced any issues (yet)

**Comment by emmggi** [comment]
@cedws @fosspill have you been using it recently? Were you banned?

**Comment by cedws** [comment]
Nothing since, but the rate limits are heavier.

**Comment by emmggi** [comment]
Can you tell me what parameters i need to set not to go over those limits? @cedws

**Comment by cedws** [comment]
You don't need to do anything. The program will sleep for as long as the server asks.
